### PR TITLE
Fix and test JsonBinarySerializer

### LINKED ISSUE #51: Add tests for JsonBinarySerializer

### PR DESCRIPTION
**Pull request by rabuckley**
Closes #51

### CODE REVIEW

File: src/DSE.Open/Text/Json/JsonSharedOptions.cs [hunk 1]
```diff
@@ -25,21 +28,26 @@ public static class JsonSharedOptions
...
-        = Create(encoder: JavaScriptEncoders.RelaxedJsonEscaping);
+    public static readonly JsonSerializerOptions RelaxedJsonEscaping = Create(encoder: JavaScriptEncoders.RelaxedJsonEscaping);
 
     public static JsonSerializerOptions Create(
         bool writeIndented = false,
         bool addDefaultConverters = true,
         JavaScriptEncoder? encoder = null)
     {
         var options = new JsonSerializerOptions();
-        ConfigureJsonOptions(options, writeIndented, addDefaultConverters);
```
Comment (rabuckley) [inline src/DSE.Open/Text/Json/JsonSharedOptions.cs:37]: `encoder` was being ignored.

File: src/DSE.Open/Text/Json/JsonSharedOptions.cs [hunk 2]
```diff
@@ -62,16 +70,8 @@ public static class JsonSharedOptions
...
-            options,
-            JsonNamingPolicy.SnakeCaseLower,
-            writeIndented,
-            addDefaultConverters);
+        => ConfigureJsonOptions(options, s_defaultNamingPolicy, writeIndented, addDefaultConverters);
 
-    [UnconditionalSuppressMessage("Trimming", "IL2026",
-        Justification = "Calls AddDefaultConverters with includeJsonValueObjectConverter = false")]
-    [UnconditionalSuppressMessage("Trimming", "IL3050",
-        Justification = "Calls AddDseOpenCoreJsonConverters with includeJsonValueObjectConverter = false")]
```
Comment (rabuckley) [inline src/DSE.Open/Text/Json/JsonSharedOptions.cs:74]: Removed the invalid suppression.

File: src/DSE.Open/Text/Json/Serialization/JsonBinarySerializer.cs [hunk 3]
```diff
@@ -11,24 +11,25 @@ namespace DSE.Open.Text.Json.Serialization;
...
 {
+    private static readonly JsonSerializerOptions s_jsonSerializerOptions = JsonSharedOptions.RelaxedJsonEscaping;
+
     [RequiresDynamicCode(WarningMessages.RequiresDynamicCode)]
     [RequiresUnreferencedCode(WarningMessages.RequiresUnreferencedCode)]
     public static ReadOnlyMemory<byte> SerializeToUtf8Json<T>(T value, JsonSerializerOptions? jsonSerializerOptions = null)
     {
-        jsonSerializerOptions ??= JsonSharedOptions.RelaxedJsonEscaping;
-        var bufferWriter = new ArrayBufferWriter<byte>();
-        using var jsonWriter = new Utf8JsonWriter(bufferWriter);
```
Comment (rabuckley) [inline src/DSE.Open/Text/Json/Serialization/JsonBinarySerializer.cs:22]: The `encoder` needs to be set on `jsonWriter` to be used. We were not passing any options, so it was using defaults (even though options are passed to the `Serialize` method.

File: src/DSE.Open/Text/Json/Serialization/JsonBinarySerializer.cs [hunk 4]
```diff
@@ -11,24 +11,25 @@ namespace DSE.Open.Text.Json.Serialization;
...
-        JsonSerializer.Serialize(jsonWriter, value, jsonSerializerOptions);
-        return bufferWriter.WrittenMemory;
+        jsonSerializerOptions ??= s_jsonSerializerOptions;
+        return JsonSerializer.SerializeToUtf8Bytes(value, jsonSerializerOptions);
```
Comment (rabuckley) [inline src/DSE.Open/Text/Json/Serialization/JsonBinarySerializer.cs:25]: Removed the usage of `Utf8JsonWriter` anyway, and used the provided `SerializeToUtf8Bytes`

**Review by frankbuckley — APPROVED** [review]

**Pull request merged**